modules/services/spotifyd.nix

```diff
@@ -0,0 +1,184 @@
+{ config, lib, pkgs, ... }:
+
+let
+  cfg = config.services.spotifyd;
+
+  configFile = pkgs.writeText "spotifyd.conf" ''
+    [global]
+    username = ${cfg.user}
+    ${lib.optionalString (cfg.password != null) "password = ${cfg.password}"}
+    ${lib.optionalString (cfg.password_cmd != null) "password_cmd = \"${cfg.password_cmd}\""}
+    ${lib.optionalString cfg.keyring "use_keyring = true"}
+    backend = ${cfg.backend}
+    device = ${cfg.device}
+    ${lib.optionalString (cfg.control_device != null) "control = ${cfg.control_device}"}
+    mixer = ${cfg.mixer}
+    volume_controller = ${cfg.alsa_audio_device}
+    ${lib.optionalString (cfg.song_change_hook != null) "on_song_change_hook = \"${cfg.song_change_hook}\""}
+    device_name = ${cfg.name}
+    bitrate = ${toString cfg.bitrate}
+    cache_path = ${cfg.cache}
+    no_audio_cache = ${toString cfg.audio_cache}
+    volume_normalisation = ${toString cfg.normalize}
+    normalisation_pregain = ${toString cfg.pregain}
+    ${lib.optionalString (cfg.zeroconf != null) "zeroconf_port = ${toString cfg.zeroconf}"}
+  '';
+in with lib; {
+  options.services.spotifyd = {
+    enable = mkEnableOption "SpotifyD connect";
+
+    package = mkOption {
```

Comment:
There is only one `spotifyd` package in Nixpkgs so I would suggest simply using that directly and removing this option.

Reply:
Yes, but if I create my own package locally, I like to have this option sticking around.

Reply:
You can still override the package through an overlay. My main concern is that I don't want to have a package option for each module.

Reply:
ok, removed that option